src/CosmosEditorManager.ts

```diff
@@ -145,11 +144,11 @@ export class CosmosEditorManager {
             const showSaveWarning: boolean | undefined = vscode.workspace.getConfiguration().get(this.showSavePromptKey);
             if (showSaveWarning !== false) {
                 const message: string = `Saving 'cosmos-editor.json' will update the entity "${editor.label}" to the Cloud.`;
-                const result: MessageItem | undefined = await vscode.window.showWarningMessage(message, DialogBoxResponses.upload, DialogBoxResponses.uploadDontWarn, DialogBoxResponses.Cancel);
+                const result: MessageItem | undefined = await vscode.window.showWarningMessage(message, { modal: true }, DialogResponses.upload, DialogResponses.alwaysUpload, DialogResponses.cancel);
```

Comment:
Do you guys think this one should be modal? It was the only one I was somewhat unsure about since we're preventing the user from continuing their work and I don't know how often they save

Reply:
I'm inclined to leave it modeless

Reply:
Discussed with @fiveisprime and he doesn't have a strong opinion either - we're fine with leaving it modal-less